react-components/src/query/network/getAssetMappedPointCloudAnnotations.ts

```diff
@@ -99,40 +98,26 @@ async function getPointCloudAnnotationDmInstances(
     .map((annotation) => (annotation.data as AnnotationsBoundingVolume).instanceRef)
     .filter((instanceRef) => instanceRef !== undefined);
 
-  const uniqueMappingDmsInstances = uniq(annotationMappingDms);
-
-  const referencesWithSources = uniqueMappingDmsInstances.map((mapping) => {
-    const reference = {
-      instanceType: mapping.instanceType,
-      externalId: mapping.externalId,
-      space: mapping.space
-    };
-
-    const sources = mapping.sources;
-    return { reference, sources };
-  });
+  const uniqueMappingDmsInstances = uniqBy(annotationMappingDms, createFdmKey);
 
-  const uniqueSources = uniq(uniqueMappingDmsInstances.flatMap((mapping) => mapping.sources));
+  const source: AnnotationsView = {
+    externalId: COGNITE_ASSET_VIEW_VERSION_KEY,
+    space: CORE_DM_SPACE,
+    version: 'v1',
+    type: 'view'
+  };
 
-  const allResultLists = await Promise.all(
-    uniqueSources.map(async (source) => {
-      return await fdmSdk.getByExternalIds<AssetProperties>(
-        referencesWithSources
-          .map((ref) => {
-            if (ref.sources.includes(source)) {
-              return ref.reference;
-            }
-            return undefined;
-          })
-          .filter((ref) => ref !== undefined),
-        [source]
-      );
-    })
-  );
-  return allResultLists.flatMap((resultList) =>
-    resultList.items.map((item) => ({
-      ...item,
-      properties: item.properties[CORE_DM_SPACE][COGNITE_ASSET_VIEW_VERSION_KEY]
-    }))
+  const allResultLists = await fdmSdk.getByExternalIds<AssetProperties>(
+    uniqueMappingDmsInstances
+      .map((ref) => {
+        if (ref.sources.includes(source)) return ref;
```

Comment:
Will this work? I thought `includes` would check for reference-equality, which means this will always return false :thinking: 

Also - I think this could be simplified to a `filter` now - so e.g. `uniqueMappingDmsInstances.filter(instance => instance.sources.some(instanceSource => instanceSource.externalId === source.externalId && instanceSource.space === source.space && instanceSource.version === source.version))`

Reply:
(We should ideally have an "isSameView" function, but I won't mandate that here)